### ISSUE TITLE
CMakeLists.txt misses definition of 'print_error'

### ISSUE THREAD

**Issue by cimatosa** [LLNL/sundials#53]
When running `cmake .` (which I should not do) I get the error

```
CMake Error at CMakeLists.txt:96 (print_error):
  Unknown CMake command "print_error".
```

Inspecting line 96 yields `print_error("In-source build prohibited.")` so the actual error message is not shown.

**Comment by apigott** [comment]
I ran into a similar error when running `cmake .. -c ../cmakelists.txt` from /build. I'm horribly unfamiliar with cmake so maybe I shouldn't do that either...

I've also tried copy/pasted from the README: `cmake -DCMAKE_INSTALL_PREFIX=/mnt/shared/sundials-5.6.1/inst -DEXAMPLES_INSTALL_PATH=/mnt/shared/sundials-5.6.1/inst/examples ../src` which returns another unknown command error `Unknown CMake command "add_prefix"`. @gardner48 do you know whether these are related and if I forgot to point to a header file somewhere with all these include statements?

**Comment by balos1** [comment]
@apigott Try `cmake -DCMAKE_INSTALL_PREFIX=/mnt/shared/sundials-5.6.1/inst -DEXAMPLES_INSTALL_PATH=/mnt/shared/sundials-5.6.1/inst/examples ../` instead. On your first run of CMake, the `../` should give the path to the root of SUNDIALS, not the `src/` directory within SUNDIALS.

**Comment by apigott** [comment]
Thanks for the quick reply! That actually gets me back to the original print_error for `CMAKE_SOURCE_DIR == CMAKE_BINARY_DIR`... I should be building in the /build right?

**Comment by balos1** [comment]
@apigott Assuming you have this layout:

```
sundials
├── CMakeLists.txt
├── CONTRIBUTING.md
├── INSTALL_GUIDE.pdf
├── LICENSE
├── NOTICE
├── README.md
├── build
├── cmake
├── doc
├── examples
├── include
├── src
└── test
```

Then, from inside the `build` directory, `cmake -DCMAKE_INSTALL_PREFIX=/mnt/shared/sundials-5.6.1/inst -DEXAMPLES_INSTALL_PATH=/mnt/shared/sundials-5.6.1/inst/examples ../` should work. If it does not, delete your `build` folder, make it again, and retry the command.

**Comment by apigott** [comment]
Hm. That didn't work so I tried tar-ing again. From the beginning:

```tar xfz sundials-5.6.1.tar.gz
tar xfz ../sun_cmake.tar.gz
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/mnt/shared/sundials-5.6.1/inst -DEXAMPLES_INSTALL_PATH=/mnt/shared/sundials-5.6.1/inst/examples ../
```

doesn't work. It's an error about the config file `CMake Error: File /mnt/shared/sundials-5.6.1/src/sundials/sundials_config.in does not exist.` It's been about a week since I could work on this but I think last time I fixed that error by copy/pasting the config settings generated by ccmake. (IIRC, from [here](http://sundials.2283335.n4.nabble.com/Installation-Issues-td4653808.html))

**Comment by balos1** [comment]
@apigott Try this:

```
git clone https://github.com/LLNL/sundials.git
cd sundials
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/mnt/shared/sundials-5.6.1/inst -DEXAMPLES_INSTALL_PATH=/mnt/shared/sundials-5.6.1/inst/examples ../
``` 

Without knowing your directory layout, it is hard to say what is going wrong.

**Comment by apigott** [comment]
ETA: Wow, I feel dumb. One of the install guides I had originally looked at ([this one](http://sundials.wikidot.com/installation-cmake#toc2)) was very old and `tar xfz ../sun_cmake.tar.gz` is deprecated. Works perfectly now I think! Thanks @balos1

**Comment by gardner48** [comment]
This issue was addressed in the SUNDIALS v5.7.0 release.
